xdsl/dialects/linalg.py

```diff
@@ -717,6 +717,92 @@ def verify_(self) -> None:
                 )
 
 
+@irdl_op_definition
+class MaxOp(NamedOpBase):
+    """
+    Takes the max (signed) between two inputs, elementwise.
+
+    See external [documentation](https://mlir.llvm.org/docs/Dialects/Linalg/#linalgmax-linalgmaxop).
+    """
+
+    name = "linalg.max"
+
+    def __init__(
+        self,
+        inputs: Sequence[SSAValue],
+        outputs: Sequence[SSAValue] = (),
+        res: Sequence[Attribute] | None = None,
+        attributes: dict[str, Attribute] | None = None,
+    ):
+        if res is None:
+            result_types = tuple(output.type for output in outputs)
+        else:
+            result_types = res
+
+        arg_types = self.body_arg_types((*inputs, *outputs))
+        maxop = (
+            arith.MaximumfOp
```

Comment:
Is this a typo? Why the if b then X else X?

Reply:
Thanks, great spot! Upstream appears to use the maxsi/minsi ops internally.